src/librustc/ty/maps.rs

```diff
@@ -96,9 +96,9 @@ impl<'tcx> Value<'tcx> for Ty<'tcx> {
     }
 }
 
-pub struct CycleError<'a, 'tcx: 'a> {
+pub struct CycleError<'tcx> {
     span: Span,
-    cycle: RefMut<'a, [(Span, Query<'tcx>)]>,
+    cycle: Vec<(Span, Query<'tcx>)>,
```

Comment:
We don't have downgrading, but queries need mutable anyway so you wouldn't get anything.
Why do you want to query from the printing code? You could clone the stack and drop the `RefMut` when reporting a cycle error if you want, that's fine (since reporting itself costs anyway).